### ISSUE TITLE
FileExistError when running script on Windows 

### ISSUE THREAD

**Issue by vjerch** [IBM-Cloud/sql-query-clients#46]
In ibmcloudsql file SQLQuery.py in method get_result there are two parts: https://github.com/IBM-Cloud/sql-query-clients/blob/147ead4f5f4c400a20ffdaa52a45af8a4aebd84b/Python/ibmcloudsql/SQLQuery.py#L226 and https://github.com/IBM-Cloud/sql-query-clients/blob/147ead4f5f4c400a20ffdaa52a45af8a4aebd84b/Python/ibmcloudsql/SQLQuery.py#L245, only file name (using tempfile.NamedTemporaryFile()) is used as a relevant parameter going into ibm_s3transfer file _init_.py method download_file() that creates a new file name but same is closed after that read_parquet method is called causing Error 183 on Windows.

Suggested solution would be to call `tmpfile.close()` before `cos_client.download_file(Bucket=result_cos_bucket, Key=bucket_object, Filename=tempfilename)` on both lines above mentioned.

ibmcloudsql SQLQuery.py get_result (current sample):

 `elif result_format == "parquet":
tmpfile = tempfile.NamedTemporaryFile()
tempfilename = tmpfile.name
cos_client.download_file(Bucket=result_cos_bucket, Key=bucket_object, Filename=tempfilename)

partition_df = pd.read_parquet(tempfilename)
tmpfile.close()`

Simple script that replicates this issue:

```
import os
import sys
import errno
import tempfile

if sys.platform.startswith('win'):
    def rename_file(current_filename, new_filename):
        try:
            os.remove(new_filename)
        except OSError as e:
            if not e.errno == errno.ENOENT:
                # We only want to a ignore trying to remove
                # a file that does not exist.  If it fails
                # for any other reason we should be propagating
                # that exception.
                raise
        os.rename(current_filename, new_filename)
else:
    rename_file = os.rename


def try_create_and_write(expect_error):
    if expect_error:
        try:
            print 'Creating temp file that will cause Windows Error 183'
            tf = tempfile.NamedTemporaryFile()
            name = tf.name

            with open(name + 'suffix', 'wb') as f:
                f.write("xxx".encode())

            rename_file(name + 'suffix', tf.name)
            tf.close()
        except Exception as e:
            print 'Got expected exception: %s' % e
    else:
        print 'Creating temp file that will be closed before renaming'
        tf = tempfile.NamedTemporaryFile()
        name = tf.name

        with open(name + 'suffix', 'wb') as f:
            f.write("xxx".encode())

        tf.close()
        rename_file(name + 'suffix', tf.name)
        print 'Got expected behavior'


def main():
    try_create_and_write(expect_error=True)
    try_create_and_write(expect_error=False)


if __name__ == "__main__":
    main()
```

**Comment by barry-hueston-IBM** [comment]
Hi, could you give us some feedback on this issue please?  Has it been assigned for review?

Thanks
Barry

**Comment by torsstei** [comment]
@barry-hueston-IBM @vjerch 
I don't have a Windows env to test. But I changed the sequence as suggested and demonstrated kin your repro and tested function still is correct on Linux. Assuming and hoping that this will resolve the issue as you demonstrated in your windows repro. Merged via https://github.com/IBM-Cloud/sql-query-clients/commit/8d6d35e5b7e090f1b08a90243d5748ba97d8689d and published on pipit as v 0.3.2
